### PR TITLE
Added Audit logs to the Client Side of the Panel

### PR DESCRIPTION
**Pull request by Tdash2**
As simple as the title says.

### CODE REVIEW

**Comment by Jelcoo** [comment]
Yes basically this will break some stuff and it will not work as intended

**Comment by Tdash2** [comment]
What would you like to see changed to make this work as intended

**Comment by Tdash2** [comment]
I just went through and made a few changes, I think it should fix the issues that you bought up.

**Comment by Jelcoo** [comment]
Okay first, don't use raw database imports, use models instead. Secondly, in app/Http/Controllers/Api/Client/Servers/ClogsController.php there are a lot of unused imports. The third thing is the spaghetti code in resources/scripts/components/server/logs/logsContainer.tsx. As last, I have a question, where does the C in ClogsController stand for?

**Comment by Tdash2** [comment]
I will fix what you brought up:
> Okay first, don't use raw database imports, use models instead. Secondly, in app/Http/Controllers/Api/Client/Servers/ClogsController.php there are a lot of unused imports. The third thing is the spaghetti code in resources/scripts/components/server/logs/logsContainer.tsx.
 
> As last, I have a question, where does the C in ClogsController stand for?

a few of the addons out already have a logsController so I added the "c" in front of it to stop issues from occurring. I could changed them to clientLogsController.

**Comment by Jelcoo** [comment]
Maybe use AuditLogsController

**Comment by FoksVHox** [comment]
> I will fix what you brought up:
> 
> > Okay first, don't use raw database imports, use models instead. Secondly, in app/Http/Controllers/Api/Client/Servers/ClogsController.php there are a lot of unused imports. The third thing is the spaghetti code in resources/scripts/components/server/logs/logsContainer.tsx.
> 
> > As last, I have a question, where does the C in ClogsController stand for?
> 
> a few of the addons out already have a logsController so I added the "c" in front of it to stop issues from occurring. I could changed them to clientLogsController.

I don't think that would be the best solution. As of the current state, Pterodactyl doesn't offer official support for add-ons, and therefore Pterodactyl should not take these addons into account. IMO it would be best to remove the "C" completely.

**Comment by trenutoo** [comment]
Without commenting on the code, since you're refactoring it, naming it logs or server logs is misleading since these would be audit logs. It should follow the naming convention and stick to audit-logs for the routes, description, and controller.

File: resources/scripts/components/server/logs/logsContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                                (data.requests.map((item, key) => (
+                                    <GreyRowBox $hoverable={false} css={tw`mb-2`} key={key}>
+                                        <GreyRowBox $hoverable={false} key={key}>
+                                            <div css={tw`flex items-center w-full md:w-auto`}>          
+                                                <div css={tw`flex-initial ml-16 text-center`}>
+                                                    <p css={tw`text-sm`}>{item.user_id}</p>
+                                                    <p css={tw`mt-1 text-2xs text-neutral-300 uppercase select-none`}>User</p>
+                                                </div>
+                                                <div css={tw`flex-initial ml-4 text-center`}>
+                                                    {item.action === 'server:filesystem.download' ?
```
Comment (schrej) [inline resources/scripts/components/server/logs/logsContainer.tsx:58]: Please use a dictionary to map an action to a text representation. This is unreadable.

**Review by schrej — CHANGES_REQUESTED** [review]

File: resources/scripts/components/server/logs/logsContainer.tsx (hunk 1 above)
Comment (Tdash2) [reply resources/scripts/components/server/logs/logsContainer.tsx:58]: > The third thing is the spaghetti code in resources/scripts/components/server/logs/logsContainer.tsx.

I am assuming this is what Jelcoo was talking about.

**Comment by FoksVHox** [comment]
This should really be superseeded by #3856

**Comment by Tdash2** [comment]
> This should really be superseeded by #3856

Yes It will. I will just close this request. Thanks for all the input

**Pull request closed without merging**